rotkehlchen/tasks/calendar.py

```diff
@@ -47,3 +70,96 @@ def delete_past_calendar_entries(database: DBHandler) -> None:
             'DELETE FROM calendar WHERE timestamp < ? AND auto_delete=1',
             (ts_now(),),
         )
+
+
+def maybe_create_ens_reminders(database: DBHandler) -> None:
+    """Check ENS registration and renewal history events and create reminders if needed."""
+    db_calendar = DBCalendar(database=database)
+    db_history_events = DBHistoryEvents(database=database)
+    customizable_date = CustomizableDateMixin(database=database)
+    ens_events: list['EvmEvent'] = []
+    with database.conn.read_ctx() as cursor:
+        for event_type, event_subtype in (
+            (HistoryEventType.TRADE, HistoryEventSubType.SPEND),
+            (HistoryEventType.RENEW, HistoryEventSubType.NONE),
+        ):
+            ens_events.extend(db_history_events.get_history_events(
+                cursor=cursor,
+                has_premium=True,  # to not hit the free limit while internal processing
+                group_by_event_ids=False,
+                filter_query=EvmEventFilterQuery.make(
+                    and_op=True,
+                    counterparties=[CPT_ENS],
+                    event_types=[event_type],
+                    event_subtypes=[event_subtype],
+                ),
+            ))
+
+    with database.conn.read_ctx() as cursor:
+        blockchain_accounts = database.get_blockchain_accounts(cursor=cursor)
+
+    current_ts = ts_now()
+    ens_to_event: dict[str, tuple[int, EvmEvent]] = {}
+    for ens_event in ens_events:
+        if not (
+            ens_event.location_label is not None and
+            ens_event.extra_data is not None and
+            'fullname' in ens_event.extra_data and
+            'expires' in ens_event.extra_data and (
```

Comment:
```suggestion
            (expires_at := ens_event.extra_data.get('expires')) is not None (
```

and you can use `expires_at` later